src/mpid/ch4/netmod/ofi/ofi_dynproc.c

```diff
@@ -208,6 +202,32 @@ static int dynproc_send_disconnect(int conn_id)
     goto fn_exit;
 }
 
+static int dynproc_grow_conn_table(int new_max)
+{
+    int mpi_errno = MPI_SUCCESS;
+
+    if (MPIDI_OFI_global.conn_mgr.max_n_conn >= new_max) {
+        goto fn_exit;
+    }
+
+    int old_max = MPIDI_OFI_global.conn_mgr.max_n_conn;
+    MPIDI_OFI_global.conn_mgr.conn_table = MPL_realloc(MPIDI_OFI_global.conn_mgr.conn_table,
```

Comment:
This is not super important to do on this pass, but do we want to plug in a utarray data structure here to handle dynamic growth?

Reply:
One of the design point is we (should) allow random disconnection during runtime, thus leaving "holes" in the table. This is facilitated with the "free_stack" added in the later commit. The code is simple enough IMO. Using `utarray` will require a different set of decorations/work-arounds, thus I don't think it will be much beneficial.